src/js/App/GlobalFilter/tagsApi.js

```diff
@@ -32,6 +22,54 @@ export function getAllTags({ search, activeTags, registeredWith } = {}, paginati
         (pagination && pagination.page) || 1,
         undefined,
         search,
-        registeredWith
+        registeredWith,
+        undefined,
+        {
+            query: generateFilter(filter)
+        }
+    );
+}
+
+export function getAllSIDs({ activeTags, registeredWith } = {}, pagination = {}) {
+    const [workloads, SID, selectedTags] = flatTags(activeTags, false, true);
+    const filter = {
+        system_profile: {
+            ...workloads?.SAP?.isSelected && { sap_system: true },
+            sap_sids: SID
+        }
+    };
+    return sap.apiSystemProfileGetSapSids(
+        selectedTags, // tags
+        (pagination && pagination.perPage) || 10,
+        (pagination && pagination.page) || 1,
+        undefined, // staleness,
+        registeredWith,
+        undefined,
+        {
+            query: generateFilter(filter)
+        }
+    );
+}
+
+export function getAllWorkloads({ activeTags, registeredWith } = {}, pagination = {}) {
+    const [workloads, SID, selectedTags] = flatTags(activeTags, false, true);
+
+    const filter = {
+        system_profile: {
```

Comment:
This object structure is repeating a lot. Can you create a function that will return this shape?